boutdata/restart.py

```diff
@@ -627,20 +685,18 @@ def redistribute(npes, path="data", nxpe=None, output=".", informat=None, outfor
                 outfile.write(v, data)
             elif dimensions == ('x', 'y'):
                 # Field2D
-                outfile.write(
-                    v, data[ix*mxsub:(ix+1)*mxsub+2*mxg, iy*mysub:(iy+1)*mysub+2*myg])
+                outfile.write(v, get_block(data))
             elif dimensions == ('x', 'z'):
                 # FieldPerp
                 yindex_global = data.attributes['yindex_global']
                 if yindex_global + myg >= iy*mysub and yindex_global + myg < (iy+1)*mysub+2*myg:
                     outfile.write(v, data[ix*mxsub:(ix+1)*mxsub+2*mxg, :])
                 else:
-                    nullarray = BoutArray(np.zeros([mxsub+2*mxg, mysub+2*myg]), attributes={"bout_type":"FieldPerp", "yindex_global":-myg-1})
+                    nullarray = BoutArray(np.zeros([mxsub+2*mxg, mzsub]), attributes={"bout_type":"FieldPerp", "yindex_global":-myg-1})
```

Comment:
Was `mysub+2*myg` a bug in the end?

Reply:
Sorry, my original comment wasn't written right: it's definitely a bug (in the sense that `mzsub` is correct and `mysub+2*myg` is incorrect - was probably a copy-paste error when I originally wrote the code), I just don't know if it ever actually had any effect...